File: easy/e9/golang/solution.go
package e9

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}

	orig := x
	last := 0
	num := 0

	for {
		if x == 0 {
			break
		}

		last = x % 10
		if num == 0 && last == 0 {
			return false
		}

		num = (num * 10) + last

		x = x / 10
	}

	if num == orig {
		return true
	}

	return false
}

File: easy/e9/cpp/solution.cpp
#include <iostream>

class Solution {
   public:
    bool isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        int64_t palindrome = 0;
        auto orig = x;

        while (true) {
            if (palindrome > 0) {
                palindrome = palindrome * 10;
            }

            palindrome += x % 10;
            x /= 10;

            if (x == 0) {
                break;
            }
        }

        if (palindrome == orig) {
            return true;
        }

        return false;
    }
};